modules/smtp/functions.php

```diff
@@ -61,3 +61,92 @@ function get_reply_type($request) {
         return false;
     }
 }
+
+/**
+ * @subpackage smtp/functions
+ */
+if (!hm_exists('send_scheduled_message')) {
+function send_scheduled_message($handler, $mailbox, $folder, $msg_id, $send_now = false) {    
```

Comment:
Why should an unsent message go through the IMAP server instead of being stored on the user's server (e.g., database or session)? Considering that each request to the IMAP server is time-consuming, it would be far more efficient to handle such actions locally, eliminating the need for a connection to an external server.

@kroky do you have an opinion?

Reply:
Keeping it on the IMAP/JMAP/EWS server has the benefit of user being able to reuse the email if they use another webmail client or a normal mail client. It is usually preferred to store data on the server and keep cypht-stored data to a minimum as it is designed to be a webmail.

Reply:
I don't think we can count them as Cypht-only. It is a normal email scheduled to be sent. Scheduling logic remains in Cypht but user can actually send it from another client if that's more convenient for them. Also, imagine they access the same mailbox with multiple cypht webmail instances - if we store in local cypht, we again loose the functionality. It doesn't really make sense to me to keep a scheduled email locally.

Reply:
Make sense, let's keep it that way.